pgx/src/spi.rs

```diff
@@ -129,69 +150,54 @@ pub struct SpiHeapTupleData {
 }
 
 impl Spi {
-    pub fn get_one<A: FromDatum + IntoDatum>(query: &str) -> Option<A> {
-        Spi::connect(|client| {
-            let result = client.select(query, Some(1), None).first().get_one();
-            Ok(result)
-        })
+    pub fn get_one<A: FromDatum + IntoDatum + 'static>(query: &str) -> Result<Option<A>, Error> {
+        Spi::connect(|client| client.select(query, Some(1), None).first().get_one())
     }
 
-    pub fn get_two<A: FromDatum + IntoDatum, B: FromDatum + IntoDatum>(
+    pub fn get_two<A: FromDatum + IntoDatum + 'static, B: FromDatum + IntoDatum + 'static>(
         query: &str,
-    ) -> (Option<A>, Option<B>) {
-        Spi::connect(|client| {
-            let (a, b) = client.select(query, Some(1), None).first().get_two::<A, B>();
-            Ok(Some((a, b)))
-        })
-        .unwrap()
+    ) -> Result<(Option<A>, Option<B>), Error> {
+        Spi::connect(|client| client.select(query, Some(1), None).first().get_two::<A, B>())
     }
 
     pub fn get_three<
-        A: FromDatum + IntoDatum,
-        B: FromDatum + IntoDatum,
-        C: FromDatum + IntoDatum,
+        A: FromDatum + IntoDatum + 'static,
+        B: FromDatum + IntoDatum + 'static,
+        C: FromDatum + IntoDatum + 'static,
```

Comment:
I am not sure we want to add the `'static` constraint to all of these.

Reply:
This is coming from `try_from_datum` which gets it from `is_pass_by_value()`

Reply:
...which gets it from `TypeIf::of`

Reply:
Agree with @workingjubilee, this is definitely not what we want here.

Sounds like the problem is a little deeper in the stack?  I see the problem, but not a solution yet...

Oh wait...

Rather than going by Rust's TypeId seems we ought to be looking at Postgres OIDs in `IntoDatum::is_pass_by_value()`.  Thoughts?

Reply:
Fixed